### PR TITLE
apply fixes for issues spotted by coverity

### PR DESCRIPTION
**Pull request by barnasm1**
### Details:
 - use std::move
 - use const auto& where possible

### Tickets:
 - [*CVS-141675*](https://jira.devtools.intel.com/browse/CVS-141675)

### CODE REVIEW

File: src/inference/include/openvino/runtime/properties.hpp [hunk 1]
```diff
@@ -940,7 +940,7 @@ struct Properties : public Property<std::map<std::string, std::map<std::string,
     inline util::EnableIfAllStringAny<std::pair<std::string, Any>, Properties...> operator()(
         const std::string& device_name,
         Properties&&... configs) const {
-        return {name() + std::string("_") + device_name, AnyMap{std::pair<std::string, Any>{configs}...}};
+        return {name() + std::string("_") + device_name, AnyMap{std::pair<std::string, Any>{std::forward<Properties>(configs)}...}};
```
Comment (praasz) [inline src/inference/include/openvino/runtime/properties.hpp]:
```suggestion
        return {name() + std::string("_") + device_name, AnyMap{std::pair<std::string, Any>{std::move(configs)}...}};
```

File: src/inference/src/dev/exec_model_info.cpp [hunk 2]
```diff
@@ -18,8 +18,8 @@ std::shared_ptr<ov::Node> ov::exec_model_info::ExecutionNode::clone_with_new_inp
 
     cloned->set_arguments(inputs);
 
-    for (auto kvp : get_rt_info())
-        cloned->get_rt_info()[kvp.first] = kvp.second;
+    for (const auto& [key, val] : get_rt_info())
```
Comment (praasz) [inline src/inference/src/dev/exec_model_info.cpp:21]: This probably will not work

File: src/inference/src/model_reader.cpp [hunk 3]
```diff
@@ -86,16 +86,16 @@ void update_v10_model(std::shared_ptr<ov::Model>& model, bool frontendMode = fal
...
             for (const auto& param : model->get_parameters()) {
-                auto param_name = param->get_friendly_name();
+                const auto& param_name = param->get_friendly_name();
                 OPENVINO_ASSERT(leaf_names.find(param_name) == leaf_names.end() ||
                                     param->output(0).get_names().find(param_name) != param->output(0).get_names().end(),
                                 "Model operation names have collisions with tensor names.",
                                 " Please use MO to generate new IR version, it should allow to avoid the issue");
                 leaf_names.emplace(param_name, nullptr);
-                param->output(0).get_tensor().add_names({param_name});
+                param->output(0).get_tensor().add_names({std::move(param_name)});
```
Comment (praasz) [inline src/inference/src/model_reader.cpp:98]: Move will not work with constant object `const auto&`.

File: src/inference/src/os/lin/lin_system_conf.cpp [hunk 4]
```diff
@@ -80,7 +80,7 @@ CPU::CPU() {
                 }
                 std::string cache_info;
                 std::getline(cache_file, cache_info);
-                one_info[n] = cache_info;
```
Comment (praasz) [inline src/inference/src/os/lin/lin_system_conf.cpp:83]: There is more like this pattern in this file. Check other if should be corrected also.

File: src/inference/src/os/lin/lin_system_conf.cpp [hunk 5]
```diff
@@ -516,7 +516,7 @@ void parse_cache_info_linux(const std::vector<std::vector<std::string>> system_i
     if ((node_info_table.size() == 0) || (node_info_table.size() == (unsigned)_sockets)) {
         if (_sockets > 1) {
             _proc_type_table.push_back(_proc_type_table[0]);
-            _proc_type_table[0] = line_value_0;
+            _proc_type_table[0] = std::move(line_value_0);
```
Comment (praasz) [inline src/inference/src/os/lin/lin_system_conf.cpp:523]: `line_value_0` is const now, move will not work.

File: src/common/transformations/src/transformations/common_optimizations/mul_fake_quantize_fusion.cpp
Comment (praasz) [inline src/common/transformations/src/transformations/common_optimizations/mul_fake_quantize_fusion.cpp:1]: Ticket is related to inference component the transformations should not be modified.
The changes will not solve reported copy issue.

File: src/inference/include/openvino/runtime/properties.hpp [hunk 6]
```diff
@@ -940,7 +940,8 @@ struct Properties : public Property<std::map<std::string, std::map<std::string,
     inline util::EnableIfAllStringAny<std::pair<std::string, Any>, Properties...> operator()(
         const std::string& device_name,
         Properties&&... configs) const {
-        return {name() + std::string("_") + device_name, AnyMap{std::pair<std::string, Any>{configs}...}};
+        return {name() + std::string("_") + device_name,
+                AnyMap{std::pair<std::string, Any>{std::forward<Properties>(configs)}...}};
```
Comment (praasz) [inline src/inference/include/openvino/runtime/properties.hpp:944]:
```suggestion
                AnyMap{std::pair<std::string, Any>{std::move(configs)}...}};
```

**Review by praasz — CHANGES_REQUESTED** [review]

File: src/common/transformations/src/transformations/common_optimizations/mul_fake_quantize_fusion.cpp
Comment (barnasm1) [reply src/common/transformations/src/transformations/common_optimizations/mul_fake_quantize_fusion.cpp:1]: revert

File: src/inference/src/dev/exec_model_info.cpp (hunk 2 above)
Comment (barnasm1) [reply src/inference/src/dev/exec_model_info.cpp:21]: revert

File: src/inference/src/model_reader.cpp (hunk 3 above)
Comment (barnasm1) [reply src/inference/src/model_reader.cpp:98]: fix

File: src/inference/src/os/lin/lin_system_conf.cpp (hunk 4 above)
Comment (barnasm1) [reply src/inference/src/os/lin/lin_system_conf.cpp:83]: done

File: src/inference/include/openvino/runtime/properties.hpp (hunk 1 above)
Comment (barnasm1) [reply src/inference/include/openvino/runtime/properties.hpp]: universal reference should use std::forward

File: src/inference/src/dev/iplugin.cpp [hunk 7]
```diff
@@ -120,18 +120,18 @@ std::unordered_set<std::string> ov::get_supported_nodes(
...
             if (name != op->get_friendly_name())
                 fused_model_op_map[name] = op->get_friendly_name();
         }
-        transformed_model_op_map[op->get_friendly_name()] = op;
         if (is_node_supported(op)) {
             supported.insert(names.begin(), names.end());
         } else {
             unsupported.insert(names.begin(), names.end());
         }
+        transformed_model_op_map[op->get_friendly_name()] = std::move(op);
```
Comment (praasz) [inline src/inference/src/dev/iplugin.cpp:134]: Is move still required?

Comment (barnasm1) [reply src/inference/src/dev/iplugin.cpp:134]: fix

**Review by praasz — APPROVED** [review]

**Pull request merged**